### PR TITLE
Wr/data bugs

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
fixes various bugs with `data`
lots of the QA steps will be found in the associated WI

### What issues does this PR fix or reference?
@W-10290520@ https://github.com/forcedotcom/cli/issues/1318
@W-10290524@ https://github.com/forcedotcom/cli/issues/72
@W-5380834@

### CODE REVIEW

File: src/dataCommand.ts [hunk 1]
```diff
@@ -182,7 +182,11 @@ export abstract class DataCommand extends SfdxCommand {
         throw new Error(messages.getMessage('TextUtilMalformedKeyValuePair', [pair]));
       } else {
         const key = pair.substr(0, eqPosition);
-        constructedObject[key] = this.convertToBooleanIfApplicable(pair.substr(eqPosition + 1));
+        if (pair.includes('{') && pair.includes('}')) {
```
Comment (mshanemc) [inline src/dataCommand.ts:185]: I could see this being true but not being parseable JSON (ex: a big textArea that's referring to some code like a Case).

So it's probably a fine solution but we'll need a try/catch on the JSON.parse because it might not be, and then we'd want to return the original value?

File: test/commands/force/data/record/dataRecord.nut.ts [hunk 2]
```diff
@@ -247,4 +247,20 @@ describe('data:record commands', () => {
       expect(result).to.have.property('Bool__c', false);
     });
   });
+
+  it('will parse JSON correctly for update', () => {
```
Comment (mshanemc) [inline test/commands/force/data/record/dataRecord.nut.ts:251]: let's have a test to match the above comment--something that isn't valid json but contains curly brackets.

**Review by mshanemc — CHANGES_REQUESTED** [review]
I like having these new exemplars for the edge cases that keep popping up.

**Review by mshanemc — APPROVED** [review]

File: src/dataCommand.ts [hunk 3]
```diff
@@ -137,10 +137,10 @@ export abstract class DataCommand extends SfdxCommand {
       );
     }
 
-    return this.normalize<Record<unknown>>(records);
+    return this.normalize<jsforceRecord<unknown>>(records);
   }
 
-  protected stringToDictionary(str: string): Dictionary<string | boolean> {
+  protected stringToDictionary(str: string): Dictionary<string | boolean | Record<string, unknown>> {
```
Comment (shetzel) [inline src/dataCommand.ts:143]: For the sake of readability you might consider making a new type for this.

**Review by shetzel — APPROVED** [review]

**Comment by mshanemc** [comment]
QA notes

`force:data:soql:query -t -q "select id, Metadata from RemoteProxy"`
✅ has valid output
same, but with the description field being invalid json `"description": "Can I nest some {invalid JSON} here"`
✅ has valid output (treats the invalid json as string)
same, but the description field is actual json `"{"foo":"bar"}"`
✅ escapes it as JSON `"description": "\"{\"foo\":\"bar\"}\""`
same, but with nested JSON and no outer quotes `{"foo":"bar", "nested": {"a": 15}}`
✅ fully escaped `"description": "{\"foo\":\"bar\", \"nested\": {\"a\": 15}}"`

**Comment by mshanemc** [comment]
`force:data:soql:query -q "SELECT Amount, Id, Name, (SELECT Quantity, ListPrice,PriceBookEntry.UnitPrice, PricebookEntry.Name,PricebookEntry.product2.Family FROM OpportunityLineItems) FROM Opportunity"` with setup per WI
✅ 👎🏻 that's an awful looking table, BUT it does have all the values

Screen Shot 2022-02-07 at 12 29 05 PM
